connect/api/src/test/java/org/apache/kafka/connect/connector/ConnectorTest.java

```diff
@@ -0,0 +1,198 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.connector;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+public class ConnectorTest {
+
+    protected ConnectorContext context;
+    protected Connector connector;
+    protected AssertableConnector assertableConnector;
+
+    @Before
+    public void beforeEach() {
+        connector = createConnector();
+        context = createContext();
+        assertableConnector = (AssertableConnector) connector;
+    }
+
+    @Test
+    public void shouldInitializeContext() {
+        connector.initialize(context);
+        assertableConnector.assertInitialized();
+        assertableConnector.assertContext(context);
+        assertableConnector.assertTaskConfigs(null);
+    }
+
+    @Test
+    public void shouldInitializeContextWithTaskConfigs() {
+        List<Map<String, String>> taskConfigs = new ArrayList<>();
+        connector.initialize(context, taskConfigs);
+        assertableConnector.assertInitialized();
+        assertableConnector.assertContext(context);
+        assertableConnector.assertTaskConfigs(taskConfigs);
+    }
+
+    @Test
+    public void shouldStopAndStartWhenReconfigure() {
+        Map<String, String> props = new HashMap<>();
+        connector.initialize(context);
+        assertableConnector.assertContext(context);
+        assertableConnector.assertStarted(false);
+        assertableConnector.assertStopped(false);
+        connector.reconfigure(props);
+        assertableConnector.assertStarted(true);
+        assertableConnector.assertStopped(true);
+        assertableConnector.assertProperties(props);
+    }
+
+    protected ConnectorContext createContext() {
+        return new TestConnectorContext();
+    }
+
+    protected Connector createConnector() {
+        return new TestConnector();
+    }
+
+    public static class TestConnectorContext implements ConnectorContext {
+
+        @Override
+        public void requestTaskReconfiguration() {
+        }
+
+        @Override
+        public void raiseError(Exception e) {
+        }
+    }
+
+    public interface AssertableConnector {
+
+        void assertContext(ConnectorContext expected);
+
+        void assertInitialized();
+
+        void assertTaskConfigs(List<Map<String, String>> expectedTaskConfigs);
+
+        void assertStarted(boolean expected);
+
+        void assertStopped(boolean expected);
+
+        void assertProperties(Map<String, String> expected);
+
+    }
+
+    public static class TestConnector extends Connector implements AssertableConnector {
```

Comment:
Should we remove this class? Currently they can only be 2 types of connectors, source or sink connectors. This seems to introduce a third implementation of the `abstract Connector` class, but testing seems sufficient without it.